### ISSUE TITLE
Overriding region with command line option fails

### ISSUE THREAD

**Issue by brettcave** [neillturner/knife-cfn#16]
The command line parameter for `--region` fails when provided.

L65-68 of cfn_base.rb contains a parameter, used when setting up the connection. However, `knife cfn create --region` fails:

```
$ knife cfn create --region us-west-1 TestStack -f teststack.json
Error: invalid option: --region
```

Preceding `create` with `--region` fails with a different error. Setting regions in `knife.rb` works as expected.

**Comment by neillturner** [comment]
does: 
  knife cfn create TestStack -f teststack.json --region us-west-1
work. 
 knife itself is parsing the parameters rather than my plugin code so not clear where problem is.

**Comment by brettcave** [comment]
no, throws the same error, regardless of position (as long as it's in the correct module, i.e. after `create`). 

The parameter parsing for region is handled in `cfn_base`, not natively by knife (I think) - when the Fog `connection` is set up.

**Comment by neillturner** [comment]
code probably needs some re-work. 
 class CfnBase < Knife    ie extends knife class
while 
 module Ec2Base
does not so getting different behaviour.

**Comment by letrait** [comment]
The command cfn update works with the parameters --region and that it is because the cfn_update.rb file defines specifically all parameters (including region) while the cfn_create.rb does not .  In fact, every commands defines all the parameters except the command "create".

**Comment by neillturner** [comment]
thanks you are right, have not had much time to look at but i will update the create code and test

**Comment by neillturner** [comment]
closed fixed in 0.1.12
